katacomb/katacomb/continuum_pipeline.py

```diff
@@ -618,6 +618,16 @@ def _export_and_merge_scans(self):
         # Close merge file
         merge_uvf.close()
 
+    def _set_mfimage_runtime_default(self, mintime=7200., extra=0.5):
+        """
+        Get the total observed time (t_obs) currently selected in the
+        kat_adapter and set the MFImage:maxRealtime parameter to
+        max(mintime, t_obs*(1. + extra)).
+        """
+        t_obs = self.ka.shape[0] * self.ka.katdal.dump_period
```

Comment:
Are we sure we can't end up with an empty scan in self.ka ? (Thus throwing an exception on this direct indexing)

Reply:
self.ka.shape is derived directly from the shape of the opened katdal object with any selection applied which should always be a tuple of 3 indices (time, freq, corr_prod) even when any one of the axes has zero shape. For an 'empty' scan self.ka.shape[0] would be 0, and result in the expected behaviour.